File: 0112_path_sum/main.go
package main

type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

func hasPathSum(node *TreeNode, sum int) bool {
	if node == nil {
		return false
	}
	diff := sum - node.Val
	if diff == 0 && node.Left == nil && node.Right == nil {
		return true
	}
	return hasPathSum(node.Left, diff) || hasPathSum(node.Right, diff)
}

func main() {

}

File: 0112_path_sum/main.py
# Definition for a binary tree node.
class TreeNode:
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None

class Solution:
    def hasPathSum(self, root, sum):
        """
        :type root: TreeNode
        :type sum: int
        :rtype: bool
        """
        if root is None:
            return False
        diff = sum - root.val
        if diff == 0 and root.left is None and root.right is None:
            return True
        return self.hasPathSum(root.left, diff) or self.hasPathSum(root.right, diff)